cmsplugin_feed_ai/feed.py

```diff
@@ -0,0 +1,171 @@
+
+    @property
+    def profile_url(self):
+        """
+        URL which points to the profile related to this feed item.
+        """
+        return "%s/%s" % (self.service_url, self.profile_name)
+
+    @property
+    def permalink(self):
+        """
+        Permalink which points to the original feed object in external service.
+        """
+        if self.source == "twitter":
+            return "%s/status/%s" % (self.profile_url, self.source_id)
+        elif self.source == "facebook":
+            return "%s/posts/%s" % (
+                self.profile_url, self.source_id.split("_")[-1]
+            )
+
+    @property
+    def link(self):
+        return self.content_link or self.permalink
+
+
+def get_feed_settings(key=None, default=None):
+    feed_settings = getattr(settings, "AI_FEED_SETTINGS", {})
+    if key not in feed_settings:
+        warn("Missing %r settings in AI_FEED_SETTINGS" % key)
+        return default
+    return feed_settings.get(key, default)
+
+
+def get_feed_cache_key(service):
+    """Return a unique cache key for given service"""
+    return "cmsplugin_feed_ai_%s_feed" % service
+
+
+def _fetch_twitter_profile_feed(profile_name, limit):
+    """
+    Fetch Twitter feed for the profile defined in settings.
+    Twython Docs: http://twython.readthedocs.io/en/latest/api.html?highlight=get_user_timeline#twython.Twython.get_user_timeline  # noqa
+    Twitter Docs: https://dev.twitter.com/rest/reference/get/statuses/user_timeline
+    """
+    conf = get_feed_settings("twitter")
+    api = Twython(
+        conf.get("app_key", ""),
+        conf.get("access_token", ""),
+    )
+    return api.get_user_timeline(
+        screen_name=profile_name,
+        count=limit,
+    )
+
+
+def _fetch_facebook_profile_feed(profile_name, limit):
+    """
+    Fetch Facebook posts made by the profile defined in settings.
+    Docs: https://developers.facebook.com/docs/graph-api/reference/v2.7/user/feed
+    """
+    conf = get_feed_settings("facebook")
+    api = GraphAPI(conf.get("auth_token", ""))
+    wall = api.get_object(
+        "%s/posts" % (profile_name),
+        limit=limit,
+        fields="id,created_time,message,link,status_type",
+    )
+    return wall["data"]
+
+
+def _fetch_profile_feed(service, profile_name, limit):
+    """
+    Fetch profile feed from given service.
+    :ret type: Iterable of api dependant feed objects
+    """
+    if service == "twitter":
+        return _fetch_twitter_profile_feed(profile_name, limit)
+    elif service == "facebook":
+        return _fetch_facebook_profile_feed(profile_name, limit)
+
+
+def get_profile_feed(service, profile_name, limit=5):
+    """
+    Get display ready profile feed from given service.
+    This method uses cache to prevent excess calls to the service APIs.
+    :param service: Service identifier
+    :type service: Str
+    :ret type: List of FeedDisplayItems
+    """
+    key = get_feed_cache_key(service)
```

Comment:
Since the data you're caching is variant on all 3 parameters (`service`, `profile_name` and `limit`), the cache key should definitely be too.

Right now, if I were to call `get_profile_feed('twitter', 'akx')`, then `get_profile_feed('twitter', 'POTUS')`, I'd still get my tweets, not the president's...